### PR TITLE
Fix failure when `DataLoader(batch_size=None)` is passed

### LINKED ISSUE #10344: 1.5.0 data loader, with batch_size bug
## 🐛 Bug

When running a dataloader with batch_size None, i.e the dataset is already in batches, and error occurs on the 'predict' step

### To Reproduce

```
import torch
from torch.utils.data import DataLoader
import pytorch_lightning as pl


class FakeDataset(torch.utils.data.Dataset):
    def __len__(self):
        return 10

    def __getitem__(self, idx):
        return torch.rand((32, 10, 1))


class Model(pl.LightningModule):

    def training_step(self, x):
        return x

    def forward(self, x):
        return x

    def configure_optimizers(self):
        optimizer = torch.optim.Adam(self.parameters(), lr=0.0005)
        return optimizer


# set up dataloader
train_dataset = FakeDataset()
train_dataloader = DataLoader(train_dataset, batch_size=None)

# set up model
model = Model()

# set up trainer
trainer = pl.Trainer(gpus=0, max_epochs=1)

# predict over training set
_ = trainer.predict(model, train_dataloader)
# This makes an error
#   File "/Users/peterdudfield/miniconda3/envs/predict_pv_yield/lib/python3.9/site-packages/pytorch_lightning/trainer/data_loading.py", line 190, in _dataloader_init_kwargs_resolve_sampler
#     batch_size=batch_sampler.batch_size,
# AttributeError: 'NoneType' object has no attribute 'batch_size'
```

### Expected behavior

no error, and predict step to work

### Environment

* CUDA:
	- GPU:
	- available:         False
	- version:           None
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.0
	- pytorch-lightning: 1.5.0
	- tqdm:              4.62.0
* System:
	- OS:                Darwin
	- architecture:
		- 64bit
		- 
	- processor:         i386
	- python:            3.9.6
	- version:           Darwin Kernel Version 20.4.0: Thu Apr 22 21:46:47 PDT 2021; root:xnu-7195.101.2~1/RELEASE_X86_64

### Additional context

Sorry if this behaviour is no longer supported

### PR DESCRIPTION
**Pull request by peterdudfield**
## What does this PR do?

Fixes #10344 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes? I ran 'pytest tests/trainer/'
- [x] Did you list all the **breaking changes** introduced by this pull request? - None
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Comment by peterdudfield** [comment]
@carmocca Thanks for looking at this. 
Would you be able to help merge this? There see,s tp be one '[check](https://github.com/PyTorchLightning/pytorch-lightning/runs/4099913072?check_suite_focus=true)' that was cancelled, but I can't re trigger this.

**Comment by justusschock** [comment]
@peterdudfield This is not a problem. This was probably due to a timeout. Once we get the approvals we can merge it :)

**Review by justusschock — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**
